ddtrace/contrib/celery/signals.py

```diff
@@ -26,6 +28,10 @@ def trace_prerun(*args, **kwargs):
         log.debug('no pin found on task or task.app task_id=%s', task_id)
         return
 
+    if config.celery['distributed_tracing'] and HTTP_HEADER_TRACE_ID in (task.request.get('headers') or {}):
```

Comment:
I'm doing this dance because I don't want to re-set the context if there aren't actually any distributed tracing headers in this task. The return type of `extract` doesn't give me this information, it'll always return some form of `Context` even if it's not one it created out of the object you're trying to extract from. Is there a better way to achieve what I want here?

Reply:
https://github.com/DataDog/dd-trace-py/blob/1e87c9bdf7769032982349c4ccc0e1c2e6866a16/ddtrace/contrib/flask/patch.py#L275-L278